sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAdministrationUtil.java

```diff
@@ -0,0 +1,383 @@
+            keyVaultPermissions, roleDefinition.getAssignableScopes().stream()
+            .map(roleScope -> KeyVaultRoleScope.fromString(roleScope.toString()))
+            .collect(Collectors.toList()));
+    }
+
+    static PagedResponse<KeyVaultRoleAssignment> transformRoleAssignmentsPagedResponse(
+        PagedResponse<RoleAssignment> pagedResponse) {
+
+        List<KeyVaultRoleAssignment> keyVaultRoleAssignments = new ArrayList<>();
+
+        for (RoleAssignment roleAssignment : pagedResponse.getValue()) {
+            keyVaultRoleAssignments.add(roleAssignmentToKeyVaultRoleAssignment(roleAssignment));
+        }
+
+        return new TransformedPagedResponse<>(keyVaultRoleAssignments, pagedResponse);
+    }
+
+    static Response<KeyVaultRoleAssignment> transformRoleAssignmentResponse(Response<RoleAssignment> response) {
+        KeyVaultRoleAssignment keyVaultRoleAssignment = roleAssignmentToKeyVaultRoleAssignment(response.getValue());
+
+        return new TransformedResponse<>(keyVaultRoleAssignment, response);
+    }
+
+    static KeyVaultRoleAssignment roleAssignmentToKeyVaultRoleAssignment(RoleAssignment roleAssignment) {
+        RoleAssignmentPropertiesWithScope propertiesWithScope = roleAssignment.getProperties();
+
+        return new KeyVaultRoleAssignment(roleAssignment.getId(), roleAssignment.getName(), roleAssignment.getType(),
+            new KeyVaultRoleAssignmentProperties(propertiesWithScope.getRoleDefinitionId(),
+                propertiesWithScope.getPrincipalId(),
+                KeyVaultRoleScope.fromString(propertiesWithScope.getScope().toString())));
+    }
+
+    private static final class TransformedPagedResponse<L extends List<T>, T, U> implements PagedResponse<T> {
+        private final L output;
+        private final PagedResponse<U> pagedResponse;
+
+        TransformedPagedResponse(L output, PagedResponse<U> inputPagedResponse) {
+            this.output = output;
+            this.pagedResponse = inputPagedResponse;
+        }
+
+        @Override
+        public void close() throws IOException {
+        }
+
+        @Override
+        public IterableStream<T> getElements() {
+            return new IterableStream<>(output);
+        }
+
+        @Override
+        public String getContinuationToken() {
+            return pagedResponse.getContinuationToken();
+        }
+
+        @Override
+        public int getStatusCode() {
+            return pagedResponse.getStatusCode();
+        }
+
+        @Override
+        public HttpHeaders getHeaders() {
+            return pagedResponse.getHeaders();
+        }
+
+        @Override
+        public HttpRequest getRequest() {
+            return pagedResponse.getRequest();
+        }
+
+        @Override
+        public List<T> getValue() {
+            return output;
+        }
+    }
+
+    private static final class TransformedResponse<T, U> implements Response<T> {
```

Comment:
Is this class really needed? This looks to just be a copy of `SimpleResponse`: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SimpleResponse.java#L40

Reply:
This PR just refactors this class.
Will let @vcolin7 do logic changes to this class usage, in a follow up PR.